internal/inputs/http.go

```diff
@@ -112,16 +123,16 @@ func RunHTTP(dataStore *[]interface{}, doLoop *bool, yml *load.Config, api load.
 						}
 						// if it is XML, convert XML to JSON and process it
 					case load.TypeXML:
-						xmlbody := strings.NewReader(strBody)
-						jsonbody, err := xj.Convert(xmlbody)
+						xmlBody := strings.NewReader(strBody)
+						jsonBody, err := xj.Convert(xmlBody)
```

Comment:
this is quite a lot of string manipulation, probably not very good memory and cpu wise. We should try to use xml Unmarshall...

Reply:
@ardias I tried to used encoding/xml to do this task, but it cannot whereas @haihongren method does which is why he probably went down this path.

So for whatever reason, xml.Unmarshal does not function the same way as json.Unmarshal does, and cannot unmarshal into an interface{}.

Below example:
- perfectly valid XML from ->  https://www.w3schools.com/xml/note.xml
- enters the correct case for XML
- however when trying to Unmarshal to an interface{} it returns nil, with no error.

![image](https://user-images.githubusercontent.com/13043208/74332885-e2bdd880-4dea-11ea-84f0-efc4d3f500a1.png)


When using @haihongren approach it works as expected with no issue.

Reply:
Ok, to be fair I hadn't tried using xml.Unmarshall yet, I assumed it was like Json. Maybe it has different requirements. 
Let's keep it like this for now.